service/src/test/java/uk/gov/hmcts/reform/fpl/service/ApplicantMigrationServiceTest.java

```diff
@@ -0,0 +1,162 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.enums.PartyType;
+import uk.gov.hmcts.reform.fpl.model.Applicant;
+import uk.gov.hmcts.reform.fpl.model.ApplicantParty;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.OldApplicant;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {ApplicantMigrationService.class, ObjectMapper.class})
+class ApplicantMigrationServiceTest {
+
+    @Autowired
+    private ApplicantMigrationService service;
+
+    @Test
+    void shouldAddMigratedApplicantYesWhenNoApplicantData() {
+        String response = service.setMigratedValue(CaseData.builder().build());
+
+        assertThat(response).isEqualTo("Yes");
+    }
+
+    @Test
+    void shouldAddMigratedApplicantYesWhenNewApplicantExists() {
+        CaseData caseData = CaseData.builder()
+            .applicants(ImmutableList.of(Element.<Applicant>builder()
+                .value(Applicant.builder()
+                    .party(ApplicantParty.builder().build())
+                    .build())
+                .build()))
+            .build();
+
+        String response = service.setMigratedValue(caseData);
+
+        assertThat(response).isEqualTo("Yes");
+    }
+
+    @Test
+    void shouldAddMigratedApplicantNoWhenOldApplicantExists() {
+        CaseData caseData = CaseData.builder()
+            .applicant(OldApplicant.builder().build())
+            .build();
+
+        String response = service.setMigratedValue(caseData);
+
+        assertThat(response).isEqualTo("No");
+    }
+
+    @Test
+    void shouldExpandApplicantCollectionWhenNoApplicants() {
+        CaseData caseData = CaseData.builder().build();
+
+        assertThat(caseData.getApplicants()).isNull();
+        assertThat(service.expandApplicantCollection(caseData)).hasSize(1);
+    }
+
+    @Test
+    void shouldNotExpandApplicantCollectionWhenApplicantsAlreadyHasSize1() {
+        CaseData caseData = CaseData.builder().applicants(ImmutableList.of(
+            Element.<Applicant>builder().build()))
+            .build();
+
+        assertThat(caseData.getApplicants()).hasSize(1);
```

Comment:
That test is not very good. Applicants collection will always have size one regardless whether collection already had element or not. It is possible to break `service.expandApplicantCollection` so that it always populates collection with empty element and that test will always pass.

As bare minimum we should check whether party ID is same after method execution as it was before.